app/src/main/java/com/ae/apps/tripmeter/fragments/TripsListFragment.java

```diff
@@ -103,10 +154,30 @@ public void onClick(View v) {
                 showAddTripDialog();
             }
         });
-
         return view;
     }
 
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSIONS_REQUEST_READ_CONTACTS: {
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    //TODO: Reload this Fragment.Otherwise the FAB won't work
```

Comment:
That's a good idea. :+1:  I also want to add a `fetchAllContactsAsync()` method to load the data in a background thread.

One problem with this change is a few other apps consuming this library, so this would be an API breaking change. Will take this up post Balanjar release when I need to continue with MessageCounter's Uniforn release :unicorn: 

https://github.com/midhunhk/ae-apps-library#implementation